src/context.rs

```diff
@@ -10,6 +10,211 @@ use crate::ffi::types::{c_uint, c_void, AlignedType};
 use crate::ffi::{self, CPtr};
 use crate::{Error, Secp256k1};
 
+#[cfg(not(feature = "std"))]
+mod internal {
+    use core::cell::UnsafeCell;
+    use core::hint::spin_loop;
+    use core::marker::PhantomData;
+    use core::mem::ManuallyDrop;
+    use core::ops::{Deref, DerefMut};
+    use core::ptr::NonNull;
+    use core::sync::atomic::{AtomicBool, Ordering};
+
+    use crate::ffi::types::{c_void, AlignedType};
+    use crate::{ffi, AllPreallocated, Context, Secp256k1};
+
+    const MAX_SPINLOCK_ATTEMPTS: usize = 128;
+    const MAX_PREALLOC_SIZE: usize = 16; // measured at 208 bytes on Andrew's 64-bit system
```

Comment:
This assumes upstream will never release any library that has preallocated constext size larger than this constant. To my knowledge, the upstream is not promising anything like that.

I really don't like it because people could get broken by updating secp256k1 dynamic library on their system. Would it be possible to get promise of some specific maximum size from upstream?